scene/resources/animation.cpp

```diff
@@ -2441,7 +2441,7 @@ int Animation::_find(const Vector<K> &p_keys, double p_time, bool p_backward, bo
 		}
 	}
 
-	if (p_limit) {
+	if (p_limit && middle > -1) {
```

Comment:
Do we still want the error if middle is -1?

Reply:
The error says 'found the key' but if middle is -1 then we didn't find the key (binary search has gone off the left side of the array)

Reply:
The error is specifically for if a valid key is found but it's outside the animation's range; from the usage of this method both `-1` and `len` are valid return values. With that in mind this check should cover both bounds - `p_limit && middle > -1 && middle < len`.

Reply:
Thanks for the answers!

That makes sense. And also makes sense why we would want to also check that `middle < len`

Reply:
updated